src/library/scala/math/BigInt.scala

```diff
@@ -380,7 +533,7 @@ final class BigInt(val bigInteger: BigInteger)
    *  it will be converted to `Double.NEGATIVE_INFINITY` or
    *  `Double.POSITIVE_INFINITY` as appropriate.
    */
-  def doubleValue: Double = this.bigInteger.doubleValue
+  def doubleValue: Double = if (isValidInt) _long.toDouble else this.bigInteger.doubleValue
```

Comment:
Why just `isValidInt`?  Up to `1L << 52` can be converted exactly.  Even if inexact, `toDouble` should do the same thing as the `BigInteger` version.

Reply:
FIrst of all, do we agree that the behaviour of this `doubleValue` method is unchanged, i.e. it returns the same double value as the previous `BigInt` implementation?

And yes, for sure, values up to `1L << 52` can be converted exactly. I'll change that.

Reply:
Values up to `1L << 53` (not 52), and down to `-(1L << 53)`, can be converted exactly.

I believe the behavior is unchanged, yes. Unless there was something involving rounding direction that can be configured for `BigInteger`s? But I don't think so.

Reply:
> it should even be possible to go all the way to if (isValidLong) _long.toDouble

"This conversion is similar to the narrowing primitive conversion from double to float as defined in section 5.1.3 of The Java™ Language Specification"

I lack the background to interpret the specification correctly; as this concerns a performance loss in a quite narrow range of cases, I'd leave it alone. What do you think?

Reply:
@sjrd Changes about the range of exact conversions have been made. What do you think outside those values? I reckon that the range that fits in a Long with more that 53 bits is pretty narrow, so I don't feel like introducing complexity there. Is this resolved?

Reply:
@sjrd @denisrosset conclusion here?

Reply:
@denisrosset The observation from @sjrd has been taken in account.